File: Fourth Semester - Design and Programming Environments/C#/ProiectC#/model/validator/TripValidator.cs
namespace TravelAgency.model.validator;

using TravelAgency.exception;

public class TripValidator: IValidator<Trip>
{
    public bool Validate(Trip entity)
    {
        if (entity == null)
        {
            throw new ArgumentNullException(nameof(entity), "Trip cannot be null");
        }

        if (string.IsNullOrEmpty(entity.Destination))
        {
            throw new ValidationException("Destination cannot be empty");
        }

        if (string.IsNullOrEmpty(entity.TransportCompany))
        {
            throw new ValidationException("TransportCompany cannot be empty");
        }
        
        if (entity.DepartureDate == null)
        {
            throw new ValidationException("Departure Date cannot be empty");
        }

        if (entity.NumberOfAvailableSeats == null)
        {
            throw new ValidationException("Number Of Available Seats cannot be empty");
        }

        if (entity.Price == null)
        {
            throw new ValidationException("Price cannot be empty");

        }
        return true;
    }
}

File: Fourth Semester - Design and Programming Environments/Java/ProiectJava/java/org/example/modul/validator/TripValidator.java
package org.example.modul.validator;


import org.example.exception.ValidationException;
import org.example.modul.Trip;

public class TripValidator implements Validator<Trip> {
    @Override
    public void validate(Trip entity) {
        if (entity == null) {
            throw new IllegalArgumentException("Trip is null");
        }
        if (entity.getDestination() == null || entity.getDestination().isEmpty()) {
            throw new ValidationException("Destination is empty");
        }
        if (entity.getDepartureDate() == null) {
            throw new ValidationException("Departure date is null");
        }
        if (entity.getNumberOfAvailableSeats() == null) {
            throw new ValidationException("Number of available places is null");
        }
        if (entity.getPrice() < 0) {
            throw new ValidationException("Price is negative");
        }
    }
}